src/coredump/coredump.c

```diff
@@ -903,10 +902,32 @@ static int submit_coredump(
                                  coredump_size, arg_journal_size_max);
         }
 
+        /* If journald is coredumping, we have to be careful that we don't deadlock when trying to write the
+         * coredump to the journal, so we put the journal socket in nonblocking mode before trying to write
+         * the coredump to the socket. */
+
+        if (context->is_journald) {
+                jfd = journal_fd();
```

Comment:
what about having a helper to change nonblock mode, instead of a helper to get the journal fd?

Reply:
I thought about that as well, I wonder if we should make that public API or not though

Reply:
I'd say keep it private for now